src/rpc/core_rpc_server_commands_defs.h

```diff
@@ -2826,6 +2826,136 @@ namespace cryptonote
     typedef epee::misc_utils::struct_init<response_t> response;
   };
 
+  struct COMMAND_RPC_GET_N_SERVICE_NODES
+  {
+
+    // Boolean values indicate whether corresponding
+    // fields should be included in the response
+    struct included_fields {
+      bool service_node_pubkey;
+      bool registration_height;
+      bool requested_unlock_height;
+      bool last_reward_block_height;
+      bool last_reward_transaction_index;
+      bool last_uptime_proof;
+
+      bool service_node_version;
+      bool contributors;
+      bool total_contributed;
+      bool total_reserved;
+      bool staking_requirement;
+      bool portions_for_operator;
+      bool swarm_id;
+      bool operator_address;
+      bool public_ip;
+      bool storage_port;
+
+      bool block_hash;
+      bool height;
+
+      BEGIN_KV_SERIALIZE_MAP()
+      KV_SERIALIZE(service_node_pubkey)
+      KV_SERIALIZE(registration_height)
+      KV_SERIALIZE(requested_unlock_height)
+      KV_SERIALIZE(last_reward_block_height)
+      KV_SERIALIZE(last_reward_transaction_index)
+      KV_SERIALIZE(last_uptime_proof)
+      KV_SERIALIZE(service_node_version)
+      KV_SERIALIZE(contributors)
+      KV_SERIALIZE(total_contributed)
+      KV_SERIALIZE(total_reserved)
+      KV_SERIALIZE(staking_requirement)
+      KV_SERIALIZE(portions_for_operator)
+      KV_SERIALIZE(swarm_id)
+      KV_SERIALIZE(operator_address)
+      KV_SERIALIZE(public_ip)
+      KV_SERIALIZE(storage_port)
+      KV_SERIALIZE(block_hash)
+      KV_SERIALIZE(height)
+      END_KV_SERIALIZE_MAP()
+    };
+
+    using contribution = COMMAND_RPC_GET_SERVICE_NODES::response_t::contribution;
+    using contributor = COMMAND_RPC_GET_SERVICE_NODES::response_t::contributor;
+
+    struct request
+    {
+      uint32_t limit;
+      included_fields fields;
+
+      BEGIN_KV_SERIALIZE_MAP()
+      KV_SERIALIZE(limit)
+      KV_SERIALIZE(fields)
+      END_KV_SERIALIZE_MAP()
+    };
+
+    struct response
+    {
+
+      struct entry {
+        const response &parent_ref;
```

Comment:
All the information it needs is just included fields? Could use that instead of the response, and then naming it requested_fields or something makes a lot more readable to me (imo).

Reply:
changed to `requested_fields`